src/higher-order/with-design-layout-selector/index.js

```diff
@@ -0,0 +1,244 @@
+		</div>
+	)
+}
+
+const DesignLayoutSelector = props => {
+	const {
+		name,
+		layouts,
+	} = props
+	const [ designs, setDesigns ] = useState( [] )
+	const [ isBusy, setIsBusy ] = useState( true )
+	const selectedLayout = props.attributes.design
+
+	useEffect( () => {
+		getDesigns( {
+			type: 'block',
+			block: name,
+		} ).then( designs => {
+			setDesigns( designs )
+			setIsBusy( false )
+		} )
+	}, [ name, setDesigns ] )
+
+	const label = <Fragment><Icon icon="admin-settings" />{ __( 'Pick a layout or design', i18n ) }</Fragment>
+	const classNames = classnames( 'ugb-design-layout-selector', { 'is-busy': isBusy } )
+
+	// Function for handling the switching designs.
+	const handleSwitchDesign = designData => {
+		const {
+			attributes,
+		} = designData
+
+		// Refetch the initial blocks. Include the currently added block.
+		dispatch( 'stackable/util' ).updateInitialBlocks()
+		setIsBusy( false )
+
+		// Close the layout selector
+		doAction( `stackable.design-layout-selector.${ props.clientId }`, false )
+
+		// Apply the block design.
+		applyBlockDesign( attributes, props.clientId )
+	}
+
+	return (
+		<Placeholder
+			className={ classNames }
+			label={ label }
+			instructions={ !! layouts.length && __( 'Select a variation to start with.', i18n ) }
+		>
+			{ !! layouts.length && (
+				<div className="ugb-design-layout-selector__layout-items">
+					{ ( layouts || [] ).map( layout => (
+						<LayoutDesignSelectorItem
+							className="ugb-design-layout-selector__layout-item"
+							onClick={ () => {
+								doAction( `stackable.design-layout-selector.${ props.clientId }`, false )
+								// Manually trigger the setAttributes filter.
+								const newAttributes = applyFilters( `stackable.${ name }.setAttributes`, { ...props.attributes, design: layout.value }, props )
+
+								// Check if a custom filter exists in the block.
+								if ( hasFilter( `stackable.${ name }.edit.inspector.layout.attributes` ) ) {
+									props.setAttributes( applyFilters( `stackable.${ name }.edit.inspector.layout.attributes`, { ...newAttributes, design: layout.value } ) )
+								} else {
+									props.setAttributes( { design: layout.value } )
+								}
+								dispatch( 'stackable/util' ).updateInitialBlocks()
+							} }
+							key={ layout.label }
+							active={ selectedLayout === layout.value }
+							{ ...layout }
+						/>
+					) ) }
+				</div>
+			) }
+			{ !! designs.length && (
+				<div className="ugb-design-layout-selector__design-library">
+					<div className="components-placeholder__instructions" >
+						{ !! layouts.length && __( 'Or pick from our Design Library.', i18n ) }
+						{ ! layouts.length && __( 'Select a design from our library to start with.', i18n ) }
+					</div>
+					<div className="components-placeholder__fieldset ugb-design-layout-selector__design-container">
+						<div className="ugb-design-layout-selector__design-items">
```

Comment:
Can you use `ul` here instead of a `div`? You'll have to use `li` on the items inside as well.

Please also do this to the designs area

Reply:
Update: I reverted the ul and li elements to div. the ul element offsets the li items to the right